corehq/apps/userreports/tests/test_save_errors.py

```diff
@@ -68,6 +75,21 @@ def test_missing_column(self):
         with self.assertRaises(MissingColumnWarning):
             adapter.best_effort_save(doc)
 
+    def test_non_nullable_column(self):
+        self.config.configured_indicators[0]['is_nullable'] = False
+        self.config._id = 'docs id'
+        adapter = self._get_adapter()
+        adapter.build_table()
+
+        doc = {
+            "_id": '123',
+            "domain": "domain",
+            "doc_type": "CommCareCase",
+            "name": None
+        }
+        adapter.best_effort_save(doc)
+        self.assertEqual(InvalidUCRData.objects.count(), 1)
```

Comment:
Asserting `validation_name` instead of or in addition to the count seems more robust.

Reply:
good idea: https://github.com/dimagi/commcare-hq/pull/24023/commits/53b346ff3d816cc2e637dd19003ff06505023929